src/swarm/neo/client/mixins/AllNodesRequestCore.d

```diff
@@ -0,0 +1,648 @@
+            working data is expected to contain a field of type
+            AllNodesRequestSharedWorkingData called all_nodes
+        FillPayload = type of policy instance to be called to add any required
+            fields to the initial message payload sent to the node
+        HandleStatusCode = type of policy instance to be called to validate the
+            status code received from the node. Should return true to continue
+            handling the request or false to abort
+
+*******************************************************************************/
+
+public struct AllNodesRequestInitialiser ( Request, FillPayload,
+    HandleStatusCode )
+{
+    import swarm.neo.client.RequestOnConn;
+    import swarm.neo.connection.RequestOnConnBase;
+
+    /// Connection event dispatcher.
+    public RequestOnConn.EventDispatcherAllNodes conn;
+
+    /// Request context.
+    public Request.Context* context;
+
+    /// Policy object for adding fields to the initial message payload.
+    public FillPayload fill_payload;
+
+    /// Policy object for handling the status code received from the node.
+    public HandleStatusCode handle_status_code;
+
+    /***************************************************************************
+
+        Performs the logic of request initialisation. Called by
+        AllNodesRequestHandler.
+
+        Returns:
+            true if initialisation succeeded for this request-on-conn, false to
+            abort handling of the request on this connection
+
+    ***************************************************************************/
+
+    public bool initialise ( )
+    {
+        // establishConnection() should guarantee we're already connected
+        assert(this.conn.waitForReconnect() == 0);
+
+        auto all_nodes = &this.context.shared_working.all_nodes;
+
+        // We know that the connection is up, so from now on we count this
+        // request among those which the started notification depends on.
+        all_nodes.initialising++;
+        scope ( exit )
+            all_nodes.initialising--;
+
+        // Send request info to node
+        this.conn.send(
+            ( conn.Payload payload )
+            {
+                payload.add(Request.cmd.code);
+                payload.add(Request.cmd.ver);
+                this.fill_payload(payload);
+            }
+        );
+
+        // Receive status from node and stop the request if not Ok
+        auto status = conn.receiveValue!(ubyte)();
+        if ( !this.handle_status_code(status) )
+            return false;
+
+        return true;
+    }
+
+    /***************************************************************************
+
+        Called by AllNodesRequest to reset any state required for initialisation
+        to begin again. (In this case, there is no state that needs resetting.)
+
+    ***************************************************************************/
+
+    public void reset ( )
```

Comment:
Nope. `AllNodesRequestSharedWorkingData.initialising` is reset in the `scope(exit)` of `initialise` and `AllNodesRequestSharedWorkingData.called_started_notifier` should never be reset because it must be called strictly once ever during the lifetime of the request.